### ISSUE TITLE
docker_daemon check failing

### ISSUE THREAD

**Issue by micahlmartin** [DataDog/dd-agent#2073]
There are edge cases where docker containers get created and don't assign a name. I'm not sure why it happens but I have a number of containers that are stopped and do not have a name. The check will fail when it get's [here](https://github.com/DataDog/dd-agent/commit/a20a2be098dde2d06f9ea7a83899fc9bbb85bac1#diff-bfbad3e094add643a9745b3e1fdd99c9R102) with this error:

```
Traceback (most recent call last):
  File "/opt/datadog-agent/agent/checks/__init__.py", line 745, in run
    self.check(copy.deepcopy(instance))
  File "/opt/datadog-agent/agent/checks.d/docker_daemon.py", line 253, in check
    containers_by_id = self._get_and_count_containers()
  File "/opt/datadog-agent/agent/checks.d/docker_daemon.py", line 309, in _get_and_count_containers
    container_name = container_name_extractor(container)[0]
  File "/opt/datadog-agent/agent/checks.d/docker_daemon.py", line 109, in container_name_extractor
    names = sorted(c.get('Names', []))
TypeError: 'NoneType' object is not iterable
```

There needs to be a check here to make sure that `Names` is not `None` or it should catch any exceptions that occur when iterating over the containers so it doesn't cause the entire check to fail. Currently this causing a number of my hosts to not report any docker stats at all.

**Comment by hkaj** [comment]
thanks for reporting that @micahlmartin ! Actually I was investigating that right now, it's a known bug in docker-engine: docker/docker#16706 The fix made it in 1.9, so if you can upgrade docker to that version you should not face this issue anymore. We know that many people can't upgrade yet (kubernetes users for example), so we're also working on a patch to handle this case on our end. I'll let you know when it's available.

**Comment by olivielpeau** [comment]
This issue has been fixed in #2071, which shipped with v5.6.3. Thanks again @micahlmartin for reporting it!

Closing.
